elastic/elastic-agent
# Ignore enroll cmd failing to restart daemon if running on a container

## What does this PR do?

Makes the container command to ignore the enroll command not being able to restart the daemon. There is no daemon on a container, therefore nothing to be restarted.

## Why is it important?

If the enroll command fails when the agent container is started for the 1st time to enroll on fleet-server, it'd fail and exit.

## Checklist





## How to test this PR locally

Start a agent container to enroll on fleet, it should succeed

```
docker run \
  --env FLEET_ENROLL=1 \
  --env FLEET_URL=https://fleet-url:8220/ \
  --env FLEET_ENROLLMENT_TOKEN=SOME_TOKEN \
  --env FLEET_INSERUCE=true \
  docker.elastic.co/beats/elastic-agent:8.12.0-SNAPSHOT
```

## Related issues

- Closes #3628 


## Logs

See #3628 

## Questions to ask yourself

- How are we going to support this in production? 
- How are we going to measure its adoption? 
- How are we going to debug this?
- What are the metrics I should take care of?
- ...
